tests/utils/test_authenticator.py

```diff
@@ -94,6 +94,32 @@ def test_authenticator_uses_username_only_credentials(
     assert request.headers["Authorization"] == "Basic Zm9vMDAxOg=="
 
 
+def test_authenticator_ignores_locked_keyring(
+    mock_config: Config,
+    mock_remote: None,
+    http: type[httpretty.httpretty],
+    with_locked_keyring: None,
+) -> None:
+    authenticator = Authenticator(mock_config, NullIO())
+    authenticator.request("get", "https://foo001@foo.bar/files/foo-0.1.0.tar.gz")
+    request = http.last_request()
+
+    assert request.headers["Authorization"] == "Basic Zm9vMDAxOg=="
+
+
+def test_authenticator_ignores_failing_keyring(
+    mock_config: Config,
+    mock_remote: None,
+    http: type[httpretty.httpretty],
+    with_erroneous_keyring: None,
+) -> None:
+    authenticator = Authenticator(mock_config, NullIO())
+    authenticator.request("get", "https://foo001@foo.bar/files/foo-0.1.0.tar.gz")
+    request = http.last_request()
+
+    assert request.headers["Authorization"] == "Basic Zm9vMDAxOg=="
```

Comment:
Tbh I don't know what this assert is actually testing, I just copied it from the other tests. However the `Authorization` is the same for the other tests using the same URL. Additionally I don't understand why some tests use different URLs. While the tests seem to be working I cannot be sure they are correct.

Reply:
I suppose the assert is just some sort of smoke test.

However, it seems the changed code is never called here (same for `test_authenticator_ignores_locked_keyring`) because there is a username in the URL. It would be best to check that the appropriate messages are logged so that we can be sure the relevant code is really called.